src/Console/Command/CustomRuleCommand.php

```diff
@@ -126,9 +132,144 @@ static function (string $answer): string {
+            );
+
+            return;
+        }
+
+        $phpunitXmlExists = file_exists($currentDirectory . '/phpunit.xml');
+        $phpunitXmlDistExists = file_exists($currentDirectory . '/phpunit.xml.dist');
+
+        if (! $phpunitXmlExists && ! $phpunitXmlDistExists) {
+            $this->symfonyStyle->warning(
+                'No phpunit.xml or phpunit.xml.dist found. Rector could not add the rector test suite to it'
+            );
+
+            return;
+        }
+
+        $phpunitFile = $phpunitXmlExists ? 'phpunit.xml' : 'phpunit.xml.dist';
+
+        $phpunitFilePath = $currentDirectory . '/' . $phpunitFile;
+
+        $domDocument = new DOMDocument('1.0');
+        $domDocument->preserveWhiteSpace = false;
+        $domDocument->formatOutput = true;
+        $domDocument->loadXML(FileSystem::read($phpunitFilePath));
+
+        if ($this->hasRectorTestSuite($domDocument)) {
+            $this->symfonyStyle->success(
+                'The rector test suite already exists in ' . $phpunitFilePath . ". No changes were made.\n You can run the rector tests by running: phpunit --testsuite rector"
+            );
+
+            return;
+        }
+
+        $phpunitXML = $this->updatePHPUnitXMLFile($domDocument, $phpunitFilePath);
+
+FileSystem::write($phpunitFilePath, $phpunitXML, null);
+
+        $this->symfonyStyle->success(
+            'We also update ' . $phpunitFilePath . ", to add a rector test suite.\n You can run the rector tests by running: phpunit --testsuite rector"
+        );
+    }
+
+    private function hasRectorTestSuite(DOMDocument $domDocument): bool
+    {
+        foreach ($this->getTestSuiteElements($domDocument) as $testSuiteElement) {
+            foreach ($testSuiteElement->getElementsByTagName('directory') as $directoryNode) {
+                if (! $directoryNode instanceof DOMElement) {
+                    continue;
+                }
+
+                $name = $testSuiteElement->getAttribute('name');
+                if ($name !== 'rector') {
+                    continue;
+                }
+
+                $directory = $directoryNode->textContent;
+                if ($directory === 'utils/rector/tests') {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private function updatePHPUnitXMLFile(DOMDocument $domDocument, string $phpunitFilePath): string
+    {
+        $domElement = $domDocument->createElement('testsuite');
+        $domElement->setAttribute('name', 'rector');
+
+        $rectorTestSuiteDirectory = $domDocument->createElement('directory', 'utils/rector/tests');
+        $domElement->appendChild($rectorTestSuiteDirectory);
+
+        $testsuitesElement = $domDocument->getElementsByTagName('testsuites')
+            ->item(0);
+
+        if (! $testsuitesElement instanceof DOMElement) {
+            throw new ShouldNotHappenException('No testsuites element found in ' . $phpunitFilePath);
```

Comment:
probably warning only is ok, to keep user know that it not generate test suite, but it still run success